libraries/AP_HAL_ChibiOS/Scheduler.cpp

```diff
@@ -240,6 +240,25 @@ void Scheduler::register_timer_failsafe(AP_HAL::Proc failsafe, uint32_t period_u
     _failsafe = failsafe;
 }
 
+void Scheduler::reboot_to_dfu()
+{
+    // from https://community.st.com/s/question/0D50X00009XkhAzSAJ/calling-stm32429ieval-bootloader
+
+    // Reboot_Loader PROC
+    // EXPORT Reboot_Loader
+    __asm__ __volatile__("LDR R0, =0x40023844"); // RCC_APB2ENR
```

Comment:
this all looks like it could be done with normal C code. It should also use the defines from the ChibiOS headers, using things like SYSCFG.MEMRMP to access registers.
It should also be run with interrupts by using port_disable(), and should flush caches and disable caches.

Reply:
> this all looks like it could be done with normal C code. It should also use the defines from 

Yeah, I looked at converting it into C calls.  Apparently you can run into compiler optimisation issues if you do that - so I didn't try :-)

the ChibiOS headers, using things like SYSCFG.MEMRMP to access registers.


> It should also be run with interrupts by using port_disable(), and should flush caches and disable caches.

I think there's actually a bigger issue here.  I just swiped the code from somewhere. But my reading of it is that it's essentially doing a jump into the DFU code.  So why does this work on our current F4 implementation - it shouldn't stop the IWDG from running, so on F4 it should go to DFU but then watchdog (it may be doing this on the F7 - I haven't checked).